app/models/content_page.rb

```diff
@@ -1,9 +1,22 @@
 class ContentPage < ApplicationRecord
   acts_as_tree
+
   scope :top_level, -> { where("parent_id IS NULL") }
+  scope :order_by_position, -> { order("position ASC") }
 
-  validates :title, presence: true, uniqueness: true
+  validates :title, format: { with: /\A[a-zA-Z0-9 ]+\Z/, message: "Title should only contain alphabet, numeric and space characters" }, presence: true, uniqueness: true
   validates :subtitle, presence: true
   validates :markdown, presence: true
   validates :seo, presence: true
+
+  validates :position, presence: true, format: { with: /\A\d+\z/, message: "Integer only, no sign allowed" }
```

Comment:
What happens when two objects have the same position?

Reply:
An error message should be added to the response

I need to add an rspec test for that

I also added a compound index in postgres on position and parent_id

Reply:
can you make the message more relatable like "enter a number between 1 and 99" and this regex will match only numbers between 1 and 99:

/^(100|[1-9][0-9]?)$/